packages/@webex/plugin-meetings/src/multistream/remoteMediaManager.ts

```diff
@@ -601,25 +601,33 @@ export class RemoteMediaManager extends EventsScope {
   }
 
   /**
-   * Allocates receive slots to all video panes in the current selected layout
+   *  Allocates receive slots to all active speaker video panes
+   *  in the current selected layout
    */
-  private allocateSlotsToVideoPaneGroups() {
-    this.receiveSlotAllocations = {activeSpeaker: {}, receiverSelected: {}};
-
-    this.currentLayout?.activeSpeakerVideoPaneGroups?.forEach((group) => {
-      this.receiveSlotAllocations.activeSpeaker[group.id] = {slots: []};
-
-      for (let paneIndex = 0; paneIndex < group.numPanes; paneIndex += 1) {
-        // allocate a slot from the "unused" list
-        const freeSlot = this.slots.video.unused.pop();
-
-        if (freeSlot) {
-          this.slots.video.activeSpeaker.push(freeSlot);
-          this.receiveSlotAllocations.activeSpeaker[group.id].slots.push(freeSlot);
+  private allocateSlotsToActiveSpeakerPaneGroups() {
+    this.currentLayout?.activeSpeakerVideoPaneGroups
```

Comment:
probably worth adding a comment explaining what we're doing here (why it's important to have the sorting and why we're using shift()) it's basically because we want to keep the same order of the slots between layouts - always the slots from highest priority groups from previous layout to be allocated in the same order to groups of descending priority in the new layout

Reply:
added